openshift-kube-apiserver/admission/customresourcevalidation/network/validate_network_config.go

```diff
@@ -0,0 +1,127 @@
+package network
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	kubeoptions "k8s.io/kubernetes/pkg/kubeapiserver/options"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+)
+
+const PluginName = "config.openshift.io/ValidateNetwork"
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return customresourcevalidation.NewValidator(
+			map[schema.GroupResource]bool{
+				configv1.Resource("networks"): true,
+			},
+			map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
+				configv1.GroupVersion.WithKind("Network"): networkV1{},
+			})
+	})
+}
+
+func toNetworkV1(uncastObj runtime.Object) (*configv1.Network, field.ErrorList) {
+	if uncastObj == nil {
+		return nil, nil
+	}
+
+	allErrs := field.ErrorList{}
+
+	obj, ok := uncastObj.(*configv1.Network)
+	if !ok {
+		return nil, append(allErrs,
+			field.NotSupported(field.NewPath("kind"), fmt.Sprintf("%T", uncastObj), []string{"Network"}),
+			field.NotSupported(field.NewPath("apiVersion"), fmt.Sprintf("%T", uncastObj), []string{"config.openshift.io/v1"}))
+	}
+
+	return obj, nil
+}
+
+type networkV1 struct {
+}
+
+func validateNetworkServiceNodePortRangeUpdate(obj, oldObj *configv1.Network) *field.Error {
+	var err error
+	defaultRange := kubeoptions.DefaultServiceNodePortRange
+	oldRange := &defaultRange
+	newRange := &defaultRange
+
+	oldRangeStr := oldObj.Spec.ServiceNodePortRange
+	if oldRangeStr != "" {
+		if oldRange, err = utilnet.ParsePortRange(oldRangeStr); err != nil {
+			return field.Invalid(field.NewPath("spec", "serviceNodePortRange"),
+				oldRangeStr,
+				fmt.Sprintf("failed to parse the old port range: %v", err))
```

Comment:
do we know that no customer has old, invalid data? This validation would mean that the customer cannot fix the range with wrong syntax.